validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,312 @@
+from . import structure_functions as sf
+
+log = logging.getLogger(__name__)
+
+
+class Photon:
+    """Photon class computing the photon array with the LuxQED approach."""
+
+    def __init__(self, theoryid, fiatlux_runcard, replicas_id):
+        self.theory = theoryid.get_description()
+        self.fiatlux_runcard = fiatlux_runcard
+        self.replicas_id = replicas_id
+        self.q_in = 100
+        self.q_in2 = self.q_in**2
+
+        # parameters for the alphaem running
+        self.alpha_em_ref = self.theory["alphaqed"]
+        self.qref = self.theory["Qref"]
+
+        self.set_betas()
+        self.set_thresholds_alpha_em()
+
+        # structure functions
+        self.qcd_pdfs = LHAPDFSet(fiatlux_runcard["pdf_name"], "replicas")
+
+        # TODO : maybe find a different name for fiatlux_dis_F2
+        path_to_F2 = theoryid.path / "fastkernel/fiatlux_dis_F2.pineappl.lz4"
+        path_to_FL = theoryid.path / "fastkernel/fiatlux_dis_FL.pineappl.lz4"
+        self.path_to_eko_photon = theoryid.path / "eko_photon.tar"
+
+        # set fiatlux
+        self.lux = {}
+        f2 = {}
+        fl = {}
+        f2lo = {}
+        for id in replicas_id:
+            f2[id] = sf.InterpStructureFunction(path_to_F2, self.qcd_pdfs.members[id])
+            fl[id] = sf.InterpStructureFunction(path_to_FL, self.qcd_pdfs.members[id])
+            f2lo[id] = sf.F2LO(self.qcd_pdfs.members[id], self.theory)
+            with tempfile.NamedTemporaryFile(mode="w") as tmp:
+                with tmp.file as tmp_file:
+                    tmp_file.write(yaml.dump(self.fiatlux_runcard))
+                self.lux[id] = fiatlux.FiatLux(tmp_file.name)
+        # we have a dict but fiatlux wants a yaml file
+        # TODO : once that fiatlux will allow dictionaries
+        # pass directly self.fiatlux_runcard
+        for id in replicas_id:
+            self.lux[id].PlugAlphaQED(self.alpha_em, self.qref)
+            self.lux[id].InsertInelasticSplitQ(
+                [
+                    self.thresh_b,
+                    self.thresh_t if self.theory["MaxNfPdf"] == 6 else 1e100,
+                ]
+            )
+            self.lux[id].PlugStructureFunctions(f2[id].fxq, fl[id].fxq, f2lo[id].fxq)
+
+        self.xgrid = XGRID
+        self.error_matrix = self.generate_error_matrix()
+
+        self.produce_interpolators()
+
+    def alpha_em(self, q):
+        r"""
+        Compute the value of alpha_em.
+
+        Parameters
+        ----------
+        q: float
+            value in which the coupling is computed
+
+        Returns
+        -------
+        alpha_em: float
+            electromagnetic coupling
+        """
+        if q < self.thresh_c:
+            nf = 3
+        elif q < self.thresh_b:
+            nf = 4
+        elif q < self.thresh_t:
+            nf = 5
+        else:
+            nf = 6
+        return self.alpha_em_nlo(q, self.alpha_thresh[nf], self.thresh[nf], nf)
+
+    def alpha_em_nlo(self, q, alpha_ref, qref, nf):
```

Comment:
It's not so clear to me why `alpha_em_nlo` and `alpha_em` are separate functions, I think that way you could also reuse the `nf` determination that is now done twice.

Reply:
Probably I should have called it `alpha_fixed_flavor`. Here I wanted to speed up the alpha determination: in fact the values of alpha at the thresholds is always the same so you don't need every time to evolve from qref to q, crossing all the thresholds, but only from the last threshold to q, given that we have precomputed all the values of alpha at the threshold values.